rotkehlchen/chain/ethereum/modules/thegraph/decoder.py

```diff
@@ -0,0 +1,184 @@
+            if (
+                event.location_label == delegator and
+                event.address == CONTRACT_THEGRAPH_STAKING and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                initial_amount_norm = event.balance.amount
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.balance = Balance(amount=stake_amount_norm)
+                event.notes = f'Delegated {stake_amount_norm} GRT to indexer {indexer}'
+                event.counterparty = CPT_THEGRAPH
+
+                # also account for the GRT burnt due to delegation tax
+                tokens_burnt = initial_amount_norm - stake_amount_norm
+                if tokens_burnt > 0:
+                    burn_event = self.base.make_event_from_transaction(
+                        transaction=context.transaction,
+                        tx_log=context.tx_log,
+                        event_type=HistoryEventType.SPEND,
+                        event_subtype=HistoryEventSubType.FEE,
+                        asset=A_GRT,
+                        balance=Balance(amount=tokens_burnt),
+                        location_label=delegator,
+                        notes=f'{tokens_burnt} GRT has been burnt as delegation tax',
+                        counterparty=CPT_THEGRAPH,
+                        address=context.tx_log.address,
+                    )
+                    return DecodingOutput(event=burn_event)
+                break
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_stake_locked(self, context: DecoderContext) -> DecodingOutput:
+        """Processes StakeDelegatedLocked event"""
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        tokens_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        lock_timeout_secs = hex_or_bytes_to_int(context.tx_log.data[64:128])
+        tokens_amount_norm = token_normalized_value(
+            token_amount=tokens_amount,
+            token=self.token,
+        )
+        # create a new informational event about undelegation and the expiring lock on tokens
+        event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_GRT,
+            balance=Balance(),
+            location_label=delegator,
+            notes=(
+                f'Undelegated {tokens_amount_norm} GRT'
+                f' from indexer {indexer}'
+                f', lock expires in {lock_timeout_secs} seconds'
```

Comment:
Doesn't this fit in one or at least two lines? Then keep it so.
Also use present tense.

```suggestion
                f'Undelegate {tokens_amount_norm} GRT from '
                f'indexer {indexer}. Lock expires in {lock_timeout_secs} seconds'
```